oops/gold_master/__init__.py

```
@@ -1147,31 +1147,19 @@ def run_tests(self):
-                # retrieved. Instead, it will use the time the file was downloaded.
-                os.rename(localpath, dated_localpath)
-                (BACKPLANE_OUTPUT_PREFIX / dated_logpath).upload()
-
-            abs_log_path = log_path.get_local_path()
-            handler = logging.FileHandler(abs_log_path)
+                
+                # Move or copy the old log to its timestamped filename using
+                # filecache-aware replacement semantics.
+                log_path.replace(dated_logpath)
+
+            handler = logging.FileHandler(logpath.get_localpath(), mode='w')
```

Ruff (0.15.12)

[error] Undefined name `logpath`

(F821)